### PR TITLE
Add definitions for Google Calendar API (gapi.calendar)

### PR DESCRIPTION
**Pull request by tkrotoff**
Add definitions for Google Calendar API (gapi.calendar)

Since I've added tests for gapi.calendar (examples taken from Google Calendar API documentation) I had to modify existing d.ts (gapi, gapi.auth2) and add others (gapi.plus, gapi.people) for the examples to work.

Most important commits:
-- "Better gapi.load"
-- "Add definitions for Google Calendar API (gapi.calendar)"

Google Calendar API is *huge*, not everything is implemented: I've typed the APIs that I'm using (same for gapi.plus and gapi.people)

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `tsc` without errors.

### CODE REVIEW

File: gapi.auth2/index.d.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ declare namespace gapi.auth2 {
      * Calls the onInit function when the GoogleAuth object is fully initialized, or calls the onFailure function if
      * initialization fails.
      */
-    then(onInit: () => any, onFailure: (reason: string) => any): any;
+    then(onInit: () => any, onFailure?: (reason: string) => any): any;
```
Comment (flawless2011) [inline gapi.auth2/index.d.ts]: What's the justification for making the failure function optional?

Comment (tkrotoff) [inline gapi.auth2/index.d.ts]: Google documentation is not precise on this kind of question and I don't have an example from Google that shows that it is optional. But empirically it should be:
- Official doc does not say it is optional: https://developers.google.com/identity/sign-in/web/reference#googleauththenoninit-onerror
- Same for gapi.client.init: https://google.github.io/closure-library/api/goog.Thenable.html althrough it is: https://developers.google.com/api-client-library/javascript/samples/samples#loading-an-api-and-making-a-request
- Google documentation about their Promise implementation says that onFulfilled and onRejected are optional: https://developers.google.com/api-client-library/javascript/features/promises like ES6 Promise: https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es5.d.ts#L1346
- Multiple examples on Internet (could not find any example from Google itself) show that it is: [1](http://stackoverflow.com/a/36995266/990356), [2](https://github.com/google/google-api-javascript-client/issues/232#issue-165357148), [3](https://github.com/google/google-api-javascript-client/issues/239#issue-175206570), [4](https://gist.github.com/Scarygami/11ed38a1a0632fe77780#file-necessary_client-html-L39)
- I do that in my code: tested and no problem

Reading all this again I'm pretty sure `onInit` callback should be optional too.

**Comment by mhegazy** [comment]
@flawless2011 is this ready to go.

File: gapi.auth2/index.d.ts (hunk 1 above)
Comment (flawless2011) [inline gapi.auth2/index.d.ts]: OK sounds good

**Comment by flawless2011** [comment]
👍

**Pull request merged**